### PR TITLE
Fix `MedianAbsoluteError` tests

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fix `MedianAbsoluteError` distributed tests to avoid results inconsistency between `np.median()` and `torch.median()

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/contrib/metrics/regression/test_median_absolute_error.py [hunk 1]
```diff
@@ -170,12 +180,22 @@ def update(engine, i):
...
 
-        np_median_absolute_error = np.median(np.abs(np_y_true - np_y_preds))
+        e_prepend = np.insert(e, 0, e[0], axis=0)
+        np_res_prepend = np.median(e_prepend)
 
-        assert pytest.approx(res) == np_median_absolute_error
+        # The results between numpy.median() and torch.median() are Inconsistant
+        # when the length of the array/tensor is even. So this is a hack to avoid that.
+        # issue: https://github.com/pytorch/pytorch/issues/1837
+        if np_y_preds.shape[0] % 2 == 0:
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_median_absolute_error.py:195]: Here we do not need that as total size is 105 ?

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @KickItLikeShika

File: tests/ignite/contrib/metrics/regression/test_median_absolute_error.py [hunk 2]
```diff
@@ -130,9 +130,19 @@ def _test(metric_device):
 
         res = m.compute()
 
-        np_median_absolute_error = np.median(np.abs(np_y - np_y_pred))
+        e = np.abs(np_y - np_y_pred)
+        np_res = np.median(e)
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/regression/test_median_absolute_error.py:134]: This line should be moved in the next `else` statement (L145)

File: tests/ignite/contrib/metrics/regression/test_median_absolute_error.py [hunk 3]
```diff
@@ -130,9 +130,19 @@ def _test(metric_device):
...
 
-        assert np_median_absolute_error == pytest.approx(res)
+        e_prepend = np.insert(e, 0, e[0], axis=0)
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/regression/test_median_absolute_error.py:136]: These 2 lines should be moved in the next `if` statement (L143)

**Comment by sdesrozis** [comment]
@KickItLikeShika left 2 minor comments

**Review by sdesrozis — APPROVED** [review]
@KickItLikeShika Thank you ! LGTM

**Pull request merged**